python/ray/serve/tests/test_deployment_graph_classmethod.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+import sys
+from typing import Union
+import starlette
+
+import ray
+from ray import serve
+from ray.serve.application import Application
+from ray.serve.api import build as build_app
+from ray.serve.deployment_graph import ClassNode, InputNode
+from ray.serve.drivers import DAGDriver
+
+
+def maybe_build(node: ClassNode, use_build: bool) -> Union[Application, ClassNode]:
+    if use_build:
+        return Application.from_dict(build_app(node).to_dict())
+    else:
+        return node
+
+
+async def json_resolver(request: starlette.requests.Request):
+    return await request.json()
+
+
+@serve.deployment
+class Counter:
+    def __init__(self, val=0):
+        self.val = val
+
+    def inc(self, inc=1):
+        self.val += inc
+
+    def get(self):
+        return self.val
+
+
+@pytest.mark.parametrize("use_build", [False, True])
+def test_two_dags_shared_instance(serve_instance, use_build):
+    """Test classmethod chain behavior is consistent across core and serve dag.
+
+    Note this only works if serve also has one replica given each class method
+    call mutates its internal state, but forming class method call chains that
+    mutate replica state should be considered anti-pattern in serve, given
+    request could be routed to different replicas each time.
+    """
+    counter = Counter.bind(0)
+
+    with InputNode() as input_1:
+        # Will be carried over to second dag if counter reused
+        counter.inc.bind(2)
+        # Only applicable to current execution
+        counter.inc.bind(input_1)
+        dag = counter.get.bind()
+        serve_dag = DAGDriver.options(route_prefix="/serve_dag").bind(
+            dag,
+            input_schema="ray.serve.tests.test_deployment_graph_classmethod.json_resolver",
```

Comment:
*facepalm*

Reply:
yeah this part needs some serious polishing if we want `serve.build` to work nicely =.= like automatically converting to string cc: @shrekris-anyscale @simon-mo